opennms-services/src/main/java/org/opennms/netmgt/collectd/CollectableService.java

```diff
@@ -354,8 +371,15 @@ private void doRun() {
             }
         }
         
+	// If we are doing strict interval, determine how long the collection
+        // has taken, so we can cut that off of the service interval
+	long diff = 0;
+        if (strictInterval) {
+            diff = System.currentTimeMillis() - m_lastScheduledCollectionTime;
+            diff = Math.max(diff, m_spec.getInterval());
```

Comment:
Should this be `Math.min` instead of `Math.max`? Otherwise diff can still be > than m_spec.getInterval(), resulting an a negative schedule time bellow.